core/token.go

```diff
@@ -225,6 +227,44 @@ func (ts TokenService) ValidateIDToken(ctx *HttpContext, idToken string) {
 	}
 }
 
+/*
+   Extract the RoleSessionName as the username which is in the 'sub' claim of the token.
+   Function should return the user ID (string containing the <username>@<tenant>) or an
+   empty string if an error of any kind.
+*/
+func (ts TokenService) ExtractUserIDFromIDToken(ctx *HttpContext, idToken string) (string) {
+	if idToken == "" {
+		ctx.Log.Err("No ID token provided.")
+		return ""
```

Comment:
shouldn't this method return (<user id>, <error>)
(it's standard Go to return (something, error) and in case of error, error will be non-nil - see  https://blog.golang.org/error-handling-and-go)